scripts/validate_translation_files.py

```diff
@@ -1,37 +1,143 @@
-        for file_name in files:
-            pofile_path = os.path.join(root, file_name)
-            if file_name.endswith('.po') and '/en/LC_MESSAGES/' not in pofile_path:
-                po_files.append(pofile_path)
-    return po_files
+    return f'{e} {traceback.format_exc()}'
+
+
+def get_translation_files(translation_directory, specific_files: List = None, allowed_types=None) -> List[Path]:
+    """
+    List all translations '*.po' and '.json' files in the specified directory.
+    If specific_files is provided, only return files that match those paths.
+    """
+    translation_files = []
+    translations_dir = Path(translation_directory)
+
+    if specific_files:
+        # Convert to absolute paths and filter
+        for file_path in specific_files:
+            file_path = Path(file_path)
+
+            if file_path.exists() and _is_valid_translation_file(file_path, allowed_types):
+                translation_files.append(file_path)
+    else:
+        # Original behavior - walk all files
+        for file_ in translations_dir.walkfiles():
+            if _is_valid_translation_file(file_, allowed_types):
+                translation_files.append(file_)
+
+    return translation_files
 
 
-def validate_translation_file(po_file):
+def _is_valid_translation_file(file_path, allowed_types=None):
+    """
+    Check if a file is a valid translation file.
     """
-    Validate a translation file and return errors if any.
+    file_str = str(file_path)
+
+    if file_str.endswith('.po') and '/en/LC_MESSAGES/' not in file_str:
+        if not allowed_types or 'po' in allowed_types:
+            return True
+
+    if file_str.endswith('.json') and not file_str.endswith('transifex_input.json'):
+        if not allowed_types or 'json' in allowed_types:
+            return True
+
+    return False
+
+
+def validate_translation_file(translation_file, error_missing_keys=False):
+    if translation_file.endswith('.po'):
+        return validate_po_translation_file(translation_file)
+    elif translation_file.endswith('.json'):
+        return validate_json_translation_file(translation_file, error_missing_keys)
+    else:
+        raise RuntimeError(f'File not supported: {translation_file}')
+
+
+def validate_json_translation_file(translation_file, error_missing_keys=False):
+    """
+    Validate translation file against source.
+    """
+    en_file = translation_file.dirname() / '../transifex_input.json'
+    required_temp_en_file_path = translation_file.dirname() / 'en.json'
+
+    if en_file.exists():
+        # Creates a .git-ignored temp. file to allow @formatjs/cli command to run
+        shutil.copyfile(en_file, required_temp_en_file_path)
+
+        is_valid = True
+        output = ""
+
+        completed_process = subprocess.run(
+            ['npx', '@formatjs/cli', 'verify', '--structural-equality',
+                                               '--source-locale=en', 'en.json', translation_file.basename()],
```

Comment:
I left a comment on the `package.json`, but if you'd rather just use `npx` then we shouldn't need a `package.json` at all, as `npx` should install everything needed to run the command automatically.